app/models/manageiq/providers/microsoft/infra_manager/vm_or_template_shared/scanning.rb

```diff
@@ -1,4 +1,18 @@
 module ManageIQ::Providers::Microsoft::InfraManager::VmOrTemplateShared::Scanning
+  extend ActiveSupport::Concern
+
+  included do
+    supports :smartstate_analysis do
+      if archived?
+        unsupported_reason_add(:smartstate_analysis,
+                               _("Smartstate Analysis cannot be performed on archived #{self.class.model_suffix} VM."))
```

Comment:
I think `self.class.description` is better for the name.
@mzazrivec how is string interpolation handled? Is something like this good for translations?

ps. actually I would just remove the class description from the reason. That makes it also easier to refactor at a later point